modules/web/src/app/cluster/details/kubeone/machine-deployment-details/component.ts

```diff
@@ -0,0 +1,141 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {AppConfigService} from '@app/config.service';
+import {ClusterService} from '@core/services/cluster';
+import {PathParam} from '@core/services/params';
+import {UserService} from '@core/services/user';
+import {Event} from '@shared/entity/event';
+import {ExternalCluster} from '@shared/entity/external-cluster';
+import {ExternalMachineDeployment} from '@shared/entity/external-machine-deployment';
+import {Member} from '@shared/entity/member';
+import {NodeMetrics} from '@shared/entity/metrics';
+import {Node} from '@shared/entity/node';
+import {GroupConfig} from '@shared/model/Config';
+import {getMachineDeploymentHealthStatus, HealthStatus} from '@shared/utils/health-status';
+import {MemberUtils, Permission} from '@shared/utils/member';
+import {forkJoin, Subject, timer} from 'rxjs';
+import {switchMap, take, takeUntil} from 'rxjs/operators';
+
+@Component({
+  selector: 'km-kubeone-machine-deployment-details',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+})
+export class KubeOneMachineDeploymentDetailsComponent implements OnInit, OnDestroy {
+  machineDeployment: ExternalMachineDeployment;
+  machineDeploymentHealthStatus: HealthStatus;
+  nodes: Node[] = [];
+  areNodesInitialized = false;
+  events: Event[] = [];
+  metrics: Map<string, NodeMetrics> = new Map<string, NodeMetrics>();
+  cluster: ExternalCluster;
+  projectID: string;
+
+  private readonly _refreshTime = 10;
+  private readonly _unsubscribe: Subject<void> = new Subject<void>();
+
+  private _machineDeploymentID: string;
+  private _isMachineDeploymentLoaded = false;
+  private _clusterID: string;
+  private _isClusterLoaded = false;
+  private _user: Member;
+  private _currentGroupConfig: GroupConfig;
+
+  constructor(
+    private readonly _activatedRoute: ActivatedRoute,
+    private readonly _appConfig: AppConfigService,
+    private readonly _userService: UserService,
+    private readonly _clusterService: ClusterService
+  ) {}
+
+  ngOnInit(): void {
+    this._clusterID = this._activatedRoute.snapshot.paramMap.get(PathParam.ClusterID);
+    this._machineDeploymentID = this._activatedRoute.snapshot.paramMap.get(PathParam.MachineDeploymentID);
+    this.projectID = this._activatedRoute.snapshot.paramMap.get(PathParam.ProjectID);
+
+    this._userService.currentUser.pipe(take(1)).subscribe(user => (this._user = user));
+
+    this._userService
+      .getCurrentUserGroup(this.projectID)
+      .subscribe(userGroup => (this._currentGroupConfig = this._userService.getCurrentUserGroupConfig(userGroup)));
+
+    timer(0, this._refreshTime * this._appConfig.getRefreshTimeBase())
```

Comment:
KubeOne HealhStatus is calculated the same way as KKP clusters so that's why `getHealthStatus` is different from external clusters.

Regarding moving common code to a shared place, I'll think about it but I don't have any clear solution in mind because I'm planning to move all 3 clusters to their own module.